x/emissions/migrations/v4/migrate_test.go

```diff
@@ -0,0 +1,524 @@
+	s.Require().NoError(err)
+	s.Require().Equal(int64(0), churningBlock)
+	s.Require().False(inFuture)
+
+	// not the same as feeRev * stake because weight is EMAd with 0
+	lowestWeight, noPrior, err := s.emissionsKeeper.GetLowestActiveTopicWeightAtBlock(s.ctx, blockHeightEnded)
+	s.Require().False(noPrior)
+	s.Require().NoError(err)
+	s.Require().True(lowestWeight.Weight.Gt(alloraMath.ZeroDec()))
+
+	activeTopicIds, err := s.emissionsKeeper.GetActiveTopicIdsAtBlock(s.ctx, blockHeightEnded)
+	s.Require().NoError(err)
+	s.Require().Len(activeTopicIds.TopicIds, 1)
+	s.Require().NotContains(activeTopicIds.TopicIds, uint64(1))
+	s.Require().Contains(activeTopicIds.TopicIds, uint64(2))
+	s.Require().NotContains(activeTopicIds.TopicIds, uint64(3))
+}
+
+func (s *EmissionsV3MigrationTestSuite) TestMigrateTopicsWithWeightDifferentEpoch() {
+	store := runtime.KVStoreAdapter(s.storeService.OpenKVStore(s.ctx))
+	cdc := s.emissionsKeeper.GetBinaryCodec()
+
+	blockHeightEnded1 := int64(100)
+	blockHeightEnded2 := int64(200)
+	blockHeightEnded3 := int64(300)
+
+	oldTopics := []oldtypes.Topic{
+		{
+			Id:             1,
+			Creator:        "creator",
+			Metadata:       "metadata",
+			LossMethod:     "lossmethod",
+			EpochLastEnded: 0,
+			EpochLength:    blockHeightEnded1,
+			GroundTruthLag: 10,
+			PNorm:          alloraMath.NewDecFromInt64(3),
+			AlphaRegret:    alloraMath.MustNewDecFromString("0.1"),
+			AllowNegative:  false,
+			Epsilon:        alloraMath.MustNewDecFromString("0.01"),
+			// InitialRegret is being reset to account for NaNs that were previously stored due to insufficient validation
+			InitialRegret:          alloraMath.MustNewDecFromString("11"),
+			WorkerSubmissionWindow: 120,
+		},
+		{
+			Id:                     2,
+			Creator:                "creator2",
+			Metadata:               "metadata2",
+			LossMethod:             "lossmethod2",
+			EpochLastEnded:         0,
+			EpochLength:            blockHeightEnded2,
+			GroundTruthLag:         20,
+			PNorm:                  alloraMath.NewDecFromInt64(3),
+			AlphaRegret:            alloraMath.MustNewDecFromString("0.1"),
+			AllowNegative:          false,
+			Epsilon:                alloraMath.MustNewDecFromString("0.01"),
+			InitialRegret:          alloraMath.MustNewDecFromString("11"),
+			WorkerSubmissionWindow: 120,
+		},
+		{
+			Id:                     3,
+			Creator:                "creator3",
+			Metadata:               "metadata3",
+			LossMethod:             "lossmethod3",
+			EpochLastEnded:         0,
+			EpochLength:            blockHeightEnded3,
+			GroundTruthLag:         30,
+			PNorm:                  alloraMath.NewDecFromInt64(3),
+			AlphaRegret:            alloraMath.MustNewDecFromString("0.1"),
+			AllowNegative:          false,
+			Epsilon:                alloraMath.MustNewDecFromString("0.01"),
+			InitialRegret:          alloraMath.MustNewDecFromString("11"),
+			WorkerSubmissionWindow: 130,
+		},
```

Comment:
Please add cases for when values are NaN. In particular `initial_regret`. 
(our topic 5 currently has this value).

Reply:
#587